src/app/core/services/auth.service.ts

```diff
@@ -72,11 +74,39 @@ export interface BiocommonsAuth0User {
 export class AuthService {
   private auth0Service = inject(Auth0Service);
   private document = inject(DOCUMENT);
+  private http = inject(HttpClient);
 
-  isAuthenticated = toSignal(this.auth0Service.isAuthenticated$, { initialValue: false });
+  isAuthenticated = toSignal(this.auth0Service.isAuthenticated$, {
+    initialValue: false,
+  });
   isLoading = toSignal(this.auth0Service.isLoading$, { initialValue: true });
-  user = toSignal(this.auth0Service.user$ as Observable<BiocommonsAuth0User | null>, { initialValue: null });
-  isAdmin = toSignal(this.isAdmin$(), { initialValue: false });
+  user = toSignal(
+    this.auth0Service.user$ as Observable<BiocommonsAuth0User | null>,
+    { initialValue: null },
+  );
+
+  isAdmin$ = this.auth0Service.isAuthenticated$.pipe(
+    switchMap((isAuth) =>
+      isAuth
+        ? this.auth0Service.getAccessTokenSilently().pipe(
+            switchMap((token) =>
+              this.http.get<{ is_admin: boolean }>(
+                `${environment.auth0.backend}/me/is-admin`,
+                { headers: { Authorization: `Bearer ${token}` } },
+              ),
+            ),
+            map((response) => response.is_admin),
+            catchError((error) => {
+              console.error('Failed to check admin status:', error);
+              return of(false);
+            }),
+          )
+        : of(false),
+    ),
+    shareReplay(1),
```

Comment:
Can we get a comment on what this does? I don't understand RxJS enough to know what `shareReplay` does